### PR TITLE
Instance logs

### PR DESCRIPTION
**Pull request by marians**
The purpose of this change is to make instance logs accessible.

The API method for logs doesn't return JSON, so it's a bit different form most others. This requires a new internal `_simple_text_get_request` method.

Logs can be accessed through the `instanceLogs()` method.

### CODE REVIEW

**Comment by marians** [comment]
This can be tested in https://github.com/giantswarm/webclient/pull/47

File: lib/client.js [hunk 1]
```diff
@@ -149,8 +149,9 @@ var GiantSwarm = (function () {
      * @param {String} [uri] API route to be called, starting with "/"
      * @param {Function} [successCallback] Callback to be called in case of success, gets response data as argument
      * @param {Function} [errorCallback] Optional callback to be called in case of an error, error object as argument
+     * @param {Object} [params] URL parameters to be added (optional)
      */
-    _simple_data_get_request: function(uri, successCallback, errorCallback) {
+    _simple_data_get_request: function(uri, successCallback, errorCallback, params) {
```
Comment (ZeissS) [inline lib/client.js]: params should be the second argument. Callbacks are normally always the last one.

File: lib/client.js [hunk 2]
```diff
@@ -174,6 +177,39 @@ var GiantSwarm = (function () {
...
+    _simple_text_get_request: function(uri, successCallback, errorCallback, params) {
+      this._validateStringParameters({uri: uri});
+      if (!successCallback) throw new this.UserException("Parameter successCallback must be given");
+      if (typeof(successCallback) !== 'function') throw new this.UserException("Parameter successCallback must be of type function");
+      if (errorCallback !== null && typeof(errorCallback) !== 'function') throw new this.UserException('errorCallback must be null or function');
+      var headers = {
+        'Authorization': 'giantswarm ' + authToken
+      }
+      if (clusterId) {
+        headers[CLUSTER_ID_HEADER] = clusterId;
```
Comment (ZeissS) [inline lib/client.js]: Why is clusterId a global variable and not read from `this`? Would make it nicer.

File: lib/client.js (hunk 1 above)
Comment (marians) [inline lib/client.js]: Even if it's optional?

File: lib/client.js [hunk 3]
```diff
@@ -161,9 +162,11 @@ var GiantSwarm = (function () {
       if (clusterId) {
         headers[CLUSTER_ID_HEADER] = clusterId;
       }
-      request
-        .get(apiEndpoint + uri)
-        .set(headers)
+      var r = request.get(apiEndpoint + uri);
+      if (params !== null) {
```
Comment (ZeissS) [inline lib/client.js]: you can make your code a lot easier here: `request.get(..).query(params || "").set(headers).end...`

Comment (ZeissS) [inline lib/client.js]: Or default the params further above to an empty string. The `ifs` feel unnecessary

File: lib/client.js [hunk 4]
```diff
@@ -504,6 +540,30 @@ var GiantSwarm = (function () {
...
+     * @param {Integer} [numLines] Number of lines to return. If null, 10 lines will be returned.
+     * @param {Function} [successCallback] Callback receiving the statistics object as argument
+     * @param {Function} [errorCallback] Optional callback to be called in case of an error, error object as argument
+     */
+    instanceLogs: function(organizationName, instanceId, numLines, successCallback, errorCallback) {
+      this._validateStringParameters({
+        organizationName: organizationName,
+        instanceId: instanceId
+      });
+      if (numLines !== null && typeof numLines !== 'number') {
```
Comment (ZeissS) [inline lib/client.js]: So `1.31415123` or `Number.NaN` is also allowed? :)

File: lib/client.js [hunk 5]
```diff
@@ -504,6 +540,30 @@ var GiantSwarm = (function () {
...
+        throw new this.UserException("Parameter 'numLines' must be of type Number");
+      }
+      var params = {};
+      if (numLines !== null) {
+        params.t = numLines;
+      }
+      var uri = '/v1/org/' + organizationName + '/instance/' + instanceId + '/logs';
```
Comment (ZeissS) [inline lib/client.js]: you may want to use `encodeURIComponent` for those URL parameters.

Comment (marians) [inline lib/client.js]: Done

File: lib/client.js (hunk 1 above)
Comment (ZeissS) [inline lib/client.js]: I would say yes.

Comment (ZeissS) [inline lib/client.js]:
```
function(uri, params, successCallback, errorCallback) {
if (arguments.length==3) {
   errorCallback = successCallback;
   successCallback = params;
   params = "";
}
```

**Comment by marians** [comment]
@ZeissS Changed. Are we happy?

**Comment by ZeissS** [comment]
I personally don't agree with the `typeof` style, but thats totally not relevant ;) LGTM

**Comment by marians** [comment]
typeof is, according to eslint, a "unary operator" and not a function. That's why I changed it.

**Pull request merged**